test/apps/app/src/testApp.ts

```diff
@@ -318,39 +316,87 @@ class TestApp {
     document.getElementById('token-error').innerText = error as string;
   }
 
-  async bootstrapProtected(): Promise<void> {
-    this.getSDKInstance();
-    const { idToken, accessToken } = this.oktaAuth.tokenManager.getTokensSync();
+  appProtectedHTML(): string {
+    const authState = this.oktaAuth.authStateManager.getAuthState();
     let content;
-    if (idToken || accessToken) {
+    if (authState?.isAuthenticated) {
       content = `
         ${homeLink(this)}
+        <div class="pure-menu">
+          <ul class="pure-menu-list actions">
+            <li class="pure-menu-item">
+              <a id="test-auth-required" onclick="testAuthRequired(event)" class="pure-menu-link">Test auth required</a>
+            </li>
+          </ul>
+        </div>
         ${logoutLink(this)}
         <hr/>
-        <strong>You are authenticated</strong>
+        <strong id="auth-status-text">You are authenticated</strong>
       `;
     } else {
-      content = `
-        ${homeLink(this)}
-        <hr/>
-        <strong>You are NOT authenticated. Login using one of these methods:</strong>
-        ${loginLinks(this, true)}
-      `;
+      if (this.isAppInitialization()) {
+        content = `
+          ${homeLink(this)}
+          <hr/>
+          <strong id="auth-status-text">You are NOT authenticated.<br />You are being redirected to sign-in page automatically...</strong>
+        `;
+      } else {
+        content = `
+          ${homeLink(this)}
+          <hr/>
+          <strong id="auth-status-text">You are NOT authenticated.<br />Sign-in again using one of these methods:</strong>
+          ${loginLinks(this, true)}
+        `;
+      }
       this.config.state = 'protected-route-' + Math.round(Math.random() * 1000);
       this.oktaAuth.setOriginalUri(this.protectedUrl, this.config.state);
     }
+    return content;
+  }
+
+  async bootstrapProtected(): Promise<void> {
+    this.getSDKInstance();
 
     // simulate SecureRoute behaviour
-    this.oktaAuth.authStateManager.subscribe((authState: AuthState) => {
-      if (!authState.isAuthenticated) {
-        this.oktaAuth.signInWithRedirect();
-      }
+    this.oktaAuth.authStateManager.subscribe(() => {
+      this.checkAuthRequired();
+    });
+    document.addEventListener('visibilitychange', () => {
+      this.checkAuthRequired();
     });
+    this.oktaAuth.start();
```

Comment:
automatically starting the service *may* cause issues in test specs since the service may continue running after the test has completed